shared/src/main/scala/mlscript/NormalForms.scala

```diff
@@ -309,7 +342,7 @@ class NormalForms extends TyperDatatypes { self: Typer =>
   
   object DNF {
     def of(lnf: LhsNf): DNF = DNF(Conjunct(lnf, SortedSet.empty, RhsBot, SortedSet.empty) :: Nil)
-    def of(bt: BaseType): DNF = DNF.of(LhsRefined(S(bt), SortedSet.empty, RecordType.empty))
+    def of(bt: BaseType): DNF = DNF.of(LhsTop & bt getOrElse (???))
```

Comment:
Let's make this cleaner:
```suggestion
    def of(bt: BaseType): DNF = DNF.of(LhsRefined(S(bt), SortedSet.empty, bt.toRecord)
```
where we could declare `toRecord` in `BaseType` and make it simply return `Top` for everything except tuples...

Reply:
This seems not possible, as `TopType` is not a subtype of `RecordType`?

Reply:
right, it's more logical if `toRecord` always returns a `RecordType`, and when the thing is not a record it just returns `RecordType(Nil)`.